tmt/steps/prepare/install.py

```diff
@@ -96,30 +121,39 @@ def prepare_sudo(self) -> None:
                 user_output.stdout,
                 user_output.stderr)
 
-        self.sudo = '' if user_output.stdout.strip() == 'root' else 'sudo '
+        return user_output.stdout.strip() != 'root'
 
-    def prepare_command(self) -> None:
+    def prepare_command(self) -> Tuple[Command, Command]:
         """ Prepare installation command"""
         raise NotImplementedError
 
     def prepare_repository(self) -> None:
         """ Configure additional repository """
         raise NotImplementedError
 
-    def list_packages(self, package_list: List[str], title: str) -> str:
+    def operation(self, subcommand: str, args: Command) -> ShellScript:
+        return ShellScript(
+            f"{self.command.to_script()} {subcommand} {self.options.to_script()} {args.to_script()}"  # noqa: E501
+        )
+
+    def perform_operation(self, subcommand: str, args: Command) -> tmt.utils.CommandOutput:
```

Comment:
TODO: find better names for `operation` and `perform_operation`.

Reply:
Don't know. The only alternative which comes to my mind is `action`. Perhaps use `action_script()` for preparing the shell script and `action()` for performing the action? Using `perform_action()` for the latter sounds good as well.

If we stay with operation, I'd recommend the rename `operation()` to `operation_script()`.

Reply:
Renamed in e6ad733dfd739df92cc42714d5cde918308dfa4b